pkg/apis/eksctl.io/v1alpha5/cloudwatch.go

```diff
@@ -1,31 +1,67 @@
 package v1alpha5
 
+import "k8s.io/apimachinery/pkg/util/sets"
+
 // ClusterCloudWatch contains config parameters related to CloudWatch
 type ClusterCloudWatch struct {
 	//+optional
 	ClusterLogging *ClusterCloudWatchLogging `json:"clusterLogging,omitempty"`
 }
 
+// Values for `CloudWatchLogging`
+const (
+	APILogging               CloudWatchLogging = "api"
+	AuditLogging             CloudWatchLogging = "audit"
+	AuthenticatorLogging     CloudWatchLogging = "authenticator"
+	ControllerManagerLogging CloudWatchLogging = "controllerManager"
+	SchedulerLogging         CloudWatchLogging = "scheduler"
+)
+
+// Valid variants are `CloudWatchLogging` constants
+type CloudWatchLogging string
+
 // ClusterCloudWatchLogging container config parameters related to cluster logging
 type ClusterCloudWatchLogging struct {
-
 	// Types of logging to enable (see [CloudWatch docs](/usage/cloudwatch-cluster-logging/#clusterconfig-examples)).
-	// To choose from: `api`, `audit`, `authenticator`, `controllerManager`, `scheduler`
 	//+optional
-	EnableTypes []string `json:"enableTypes,omitempty"`
+	EnableTypes []CloudWatchLogging `json:"enableTypes,omitempty"`
+}
+
+// SupportedCloudWatchClusterLogTypes returns all supported logging facilities
+func SupportedCloudWatchClusterLogTypes() []CloudWatchLogging {
+	return []CloudWatchLogging{
+		APILogging,
+		AuditLogging,
+		AuthenticatorLogging,
+		ControllerManagerLogging,
+		SchedulerLogging,
+	}
 }
 
-// SupportedCloudWatchClusterLogTypes retuls all supported logging facilities
-func SupportedCloudWatchClusterLogTypes() []string {
-	return []string{"api", "audit", "authenticator", "controllerManager", "scheduler"}
+// SupportedCloudWatchClusterLogTypesSet returns all supported logging facilities
+func SupportedCloudWatchClusterLogTypesSet() sets.String {
+	return sets.NewString(StringsFromCloudWatchTypes(SupportedCloudWatchClusterLogTypes())...)
+}
+
+// StringsFromCloudWatchTypes is a util function to convert lists
+func StringsFromCloudWatchTypes(types []CloudWatchLogging) []string {
```

Comment:
What are the benefits of doing this for enums? It seems they are a bit more cumbersome to implement and there still not that much type safety.

Reply:
yeah there's not much. I despise how cumbersome it is in `go` but here it's really only because it allows me to add info (here the `enum` values) to the new `type` through comments.

https://github.com/weaveworks/eksctl/blob/a635013f054fc675571a684ab212d6850b35d990/pkg/apis/eksctl.io/v1alpha5/cloudwatch.go#L20-L21

Reply:
I'm going to fix it a different way

Reply:
got rid of the type alias

Reply:
Thanks, this looks a lot better IMO. I am also ok with just having a comment with the possible values though.